### PR TITLE
ROX-23287: compliance refactor result service

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

This is a refactor just to make file size smaller and to make it easier to work on different things at the same time.  Stats vs results seemed like a logical way to differentiate.

## Checklist
- [ ] Investigated and inspected CI test results
~- [ ] Unit test and regression tests added~
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Executed the various endpoints to ensure they still worked.  Implementation, nor endpoints didn't change.  They just moved.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: proto/api/v2/compliance_results_stats_service.proto [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    option (google.api.http) = {get: "/v2/compliance/scan/stats/cluster"};
+  }
+
+  // Deprecated in favor of GetComplianceClusterStats
+  rpc GetComplianceOverallClusterStats(RawQuery) returns (ListComplianceClusterOverallStatsResponse) {
+    option (google.api.http) = {get: "/v2/compliance/scan/stats/overall/cluster"};
+  }
+
+  rpc GetComplianceClusterStats(ComplianceProfileResultsRequest) returns (ListComplianceClusterOverallStatsResponse) {
+    option (google.api.http) = {get: "/v2/compliance/results/profiles/{profile_name}/clusters"};
```
Comment (bradr5) [inline proto/api/v2/compliance_results_stats_service.proto:88]:
```suggestion
    option (google.api.http) = {get: "/v2/compliance/scan/stats/profiles/{profile_name}/clusters"};
```

**Review by SimonBaeumer — APPROVED** [review]

File: proto/api/v2/compliance_results_stats_service.proto [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  // - scan: id(s) of the compliance scan
+  // - cluster: id(s) of the cluster
+  // - profile: id(s) of the profile
+  rpc GetComplianceProfilesStats(RawQuery) returns (ListComplianceProfileScanStatsResponse) {
+    option (google.api.http) = {get: "/v2/compliance/scan/stats/profiles"};
+  }
+
+  // GetComplianceProfileCheckStats lists current stats for a specific cluster check
+  rpc GetComplianceProfileCheckStats(ComplianceProfileCheckRequest) returns (ListComplianceProfileResults) {
+    option (google.api.http) = {get: "/v2/compliance/scan/stats/profile/{profile_name}/checks/{check_name}"};
```
Comment (bradr5) [inline proto/api/v2/compliance_results_stats_service.proto:69]:
```suggestion
    option (google.api.http) = {get: "/v2/compliance/scan/stats/profiles/{profile_name}/checks/{check_name}"};
```

**Comment by dashrews78** [comment]
@bradr5 fixed those endpoints and found a couple of others in `compliance_results_service.proto` I updated as well.

File: proto/api/v2/compliance_results_service.proto [hunk 3]
```diff
@@ -215,67 +121,19 @@ service ComplianceResultsService {
...
-  }
-
-  // Deprecated in favor of GetComplianceClusterStats
-  rpc GetComplianceOverallClusterStats(RawQuery) returns (ListComplianceClusterOverallStatsResponse) {
-    option (google.api.http) = {get: "/v2/compliance/scan/stats/overall/cluster"};
-  }
-
-  rpc GetComplianceClusterStats(ComplianceProfileResultsRequest) returns (ListComplianceClusterOverallStatsResponse) {
-    option (google.api.http) = {get: "/v2/compliance/scan/stats/profiles/{profile_name}/clusters"};
+    option (google.api.http) = {get: "/v2/compliance/scan/profiles/results/{profile_name}"};
```
Comment (bradr5) [inline proto/api/v2/compliance_results_service.proto:124]:
```suggestion
    option (google.api.http) = {get: "/v2/compliance/scan/results/profiles/{profile_name}/checks"};
```
Would be a good opportunity to rename this one while we're here

**Review by SimonBaeumer — APPROVED** [review]

**Pull request merged**